### PR TITLE
Rework layout rendering

### PR DESCRIPTION
**Pull request by WojciechBarczynski**

### CODE REVIEW

File: compositor_render/src/transformations/layout/apply_layouts.wgsl [hunk 1]
```diff
@@ -18,51 +16,36 @@ struct Layout {
...
     
-    let sample = is_inside * textureSample(textures[texture_id], sampler_, input.tex_coords) 
-        + (1.0 - is_inside) * vec4<f32>(0.0, 0.0, 0.0, 0.0);
-
-    if current_layout.texture_id != -1 {
-        return sample;
-    } else {
-        return current_layout.color;
-    }
+    return is_inside * textureSample(texture, sampler_, input.tex_coords) + (1.0 - is_inside) * vec4<f32>(0.0, 0.0, 0.0, 0.0);
```
Comment (noituri) [inline compositor_render/src/transformations/layout/apply_layouts.wgsl:50]: ```(1.0 - is_inside) * vec4<f32>(0.0, 0.0, 0.0, 0.0)``` 

Isn't it always `0`?

Comment (WojciechBarczynski) [reply compositor_render/src/transformations/layout/apply_layouts.wgsl:50]: Ohh, right :) The second part can be removed entirely :) Good catch

File: compositor_render/src/transformations/layout.rs [hunk 2]
```diff
@@ -143,13 +141,21 @@ impl LayoutNode {
...
         self.params.update(params, ctx.wgpu_ctx);
 
+        let textures: Vec<Option<&NodeTexture>> = layouts
+            .iter()
+            .map(|layout| match layout.content {
+                RenderLayoutContent::Color(_) => None,
+                RenderLayoutContent::ChildNode { index, .. } => Some(
+                    sources
+                        .get(index)
+                        .expect("Invalid source index in layout")
```
Comment (wkozyra95) [inline compositor_render/src/transformations/layout.rs:151]: Not sure if panic here would be appropriate, I would log `error!` and maybe return None

File: compositor_render/src/transformations/layout/apply_layouts.wgsl [hunk 3]
```diff
@@ -18,51 +16,36 @@ struct Layout {
...
 fn fs_main(input: VertexOutput) -> @location(0) vec4<f32> {
-    let current_layout = layouts[input.layout_id];
+    let current_layout = layouts[layout_id];
 
     // sampling can't be conditional, so in case of texture_id == -1
     // sample textures[0], but ignore the result.
-    var texture_id = current_layout.texture_id;
-    if texture_id == -1 {
-        texture_id = 0;
+    if current_layout.texture_id == -1 {
```
Comment (wkozyra95) [inline compositor_render/src/transformations/layout/apply_layouts.wgsl:44]: maybe change name to is_texture, use_texture or has_texture?

**Review by wkozyra95 — APPROVED** [review]

**Review by noituri — APPROVED** [review]

**Pull request merged**